packages/aws-cdk-lib/aws-dynamodb/lib/global-table.ts

```diff
@@ -0,0 +1,993 @@
+      }
+    } else {
+      this.billingMode = BillingMode.PAY_PER_REQUEST;
+    }
+
+    this.addKey(props.partitionKey, HASH_KEY_TYPE);
+    if (props.sortKey) {
+      this.addKey(props.sortKey, RANGE_KEY_TYPE);
+    }
+
+    this.encryption = props.encryption;
+    const sseSpecification = this.encryption ? this.configureSseSpecification(this.encryption) : undefined;
+
+    props.globalSecondaryIndexes?.forEach(gsi => this.addGlobalSecondaryIndex(gsi));
+    props.localSecondaryIndexes?.forEach(lsi => this.addLocalSecondaryIndex(lsi));
+    props.replicas?.forEach(replica => this.addReplica(replica));
+
+    const resource = new CfnGlobalTable(scope, 'Resource', {
+      tableName: this.physicalName,
+      keySchema: this.keySchema,
+      attributeDefinitions: Lazy.any({ produce: () => this.attributeDefinitions }),
+      replicas: Lazy.any({ produce: () => this.replicaTables }),
+      localSecondaryIndexes: Lazy.any({ produce: () => this.localSecondaryIndexes }, { omitEmptyArray: true }),
+      globalSecondaryIndexes: Lazy.any({ produce: () => this.globalSecondaryIndexes }, { omitEmptyArray: true }),
+      billingMode: this.billingMode,
+      streamSpecification: { streamViewType: NEW_AND_OLD_IMAGES },
+      writeProvisionedThroughputSettings: this.writeProvisioning,
+      sseSpecification,
+      timeToLiveSpecification: props.timeToLiveAttribute
+        ? { attributeName: props.timeToLiveAttribute, enabled: true }
+        : undefined,
+    });
+    resource.applyRemovalPolicy(props.removalPolicy);
+
+    this.tableArn = this.getResourceArnAttribute(resource.attrArn, {
+      service: 'dynamodb',
+      resource: 'table',
+      resourceName: this.physicalName,
+    });
+    this.tableName = this.getResourceNameAttribute(resource.ref);
+    this.tableId = resource.attrTableId;
+    this.tableStreamArn = resource.attrStreamArn;
+
+    if (props.tableName) {
+      this.node.addMetadata('aws:cdk:hasPhysicalName', this.tableName);
+    }
+  }
+
+  private get localSecondaryIndexes() {
+    const localSecondaryIndexes: CfnGlobalTable.LocalSecondaryIndexProperty[] = [];
+
+    for (const localSecondaryIndex of this._localSecondaryIndexes.values()) {
+      localSecondaryIndexes.push(localSecondaryIndex);
+    }
+
+    return localSecondaryIndexes;
+  }
+
+  private get globalSecondaryIndexes() {
+    const globalSecondaryIndexes: CfnGlobalTable.GlobalSecondaryIndexProperty[] = [];
+
+    for (const globalSecondaryIndex of this._globalSecondaryIndexes.values()) {
+      globalSecondaryIndexes.push(globalSecondaryIndex);
+    }
+
+    return globalSecondaryIndexes;
+  }
+
+  private get replicaTables() {
+    const replicaTables: CfnGlobalTable.ReplicaSpecificationProperty[] = [];
+
+    if (!this._replicaTables.has(this.region)) {
+      const replicaTable = this.configureReplicaTable({ region: this.region });
+      replicaTables.push(replicaTable);
+    }
```

Comment:
Can't we make it impossible to define a replica with the current region?

What's the use case for allowing this?

Reply:
No, this is the default for a Global Table. From the CloudFormation documentation - "The list must contain at least one element, the region where the stack defining the global table is deployed. For example, if you define your table in a stack deployed to us-east-1, you must have an entry in Replicas with the region us-east-1. You cannot remove the replica in the stack region" (https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-globaltable.html#cfn-dynamodb-globaltable-replicas)

Reply:
I think I misunderstood, but you're right. I updated this.